src/eterna/mode/DesignBrowser/DataCol.ts

```diff
@@ -19,23 +19,31 @@ import {SortOrder} from './SortOptions';
 import SequenceStringListView from './SequenceStringListView';
 import {DesignBrowserDataType, DesignCategory, DBVote} from './DesignBrowserMode';
 
+interface DataColProps {
+    dataType: DesignBrowserDataType;
+    category: DesignCategory;
+    dataWidth: number;
+    fonttype: string;
+    fontSize: number;
+    sortable: boolean;
+    domParent: string | HTMLElement;
+}
 export default class DataCol extends ContainerObject {
     public readonly sortOrderChanged = new Signal<SortOrder>();
     public readonly filtersChanged = new UnitSignal();
     public readonly voteChanged = new Signal<number>();
     public readonly category: DesignCategory;
 
-    constructor(dataType: DesignBrowserDataType, category: DesignCategory,
-        dataWidth: number, fonttype: string,
-        fontSize: number, sortable: boolean) {
+    constructor(props: DataColProps) {
```

Comment:
So the real non-cosmetic change throughout here is essentially the addition of the `domParent` -- can you explain why DataCol needs to know that? Just so the TextInputObjects it creates share its parent?

Reply:
Yeah it's so we can pass it to TextInputObject, which passes it to DOMObject, which uses it to insert itself into the DOM at the correct point (specifically, as a child of the masked div created by ScrollContainer)